automatic/pycharm-edu/tools/chocolateyinstall.ps1

```diff
@@ -21,4 +21,16 @@ $packageArgs = @{
   destination    = $toolsDir
 }
 
+[array]$key = Get-UninstallRegistryKey -SoftwareName $packageArgs['softwareName']
+
+if ($key.Count -gt 0) {
+    Get-Process PyCharm* | ForEach-Object { $_.CloseMainWindow() }
+  $key | ForEach-Object {
+    $packageArgs['file'] = "$($_.UninstallString)"
+    Uninstall-ChocolateyPackage @packageArgs
+  }
+} else {
+  Write-Warning "$packageName has already been uninstalled by other means."
+}
+
```

Comment:
This is from the beforeModify file and should be there.

Reply:
Older versions should be removed on install AND upgrade, not ONLY on upgrade. Do you agree?

This is why I suggest you to invoke Uninstall-ChocolateyPackage in chocolateyinstall.ps1. So you won't have to wait an upgrade for cleaning old version. 

Furthermore, Chocolatey will look for and run a chocolateyBeforeModify.ps1 file in the EXISTING package prior to upgrading or uninstalling a package. So when installing for the first time, "cleaning process" will never run... 

In a general way, I think we must manage "cleaning process" outside chocolateybeforemodify.ps1, as it is recommended in documentation : https://chocolatey.org/docs/getting-started

In this state, we have a lot of systems with dozens of versions of pycharm because cleaning process doesn't work as expected.

I would appreciate if you reconsider my pull request.

Thanks a lot,
Guillaume

Reply:
A chocolatey package is not allowed to call uninstall commands in an install action. This should be done in the "Before modify" file. 

While I understand your thought process and I mostly agree, I am not allowed to push packages with this method.